Add links and comments to vSphere clusterspec docs (#5078)
aws/eks-anywhere: docs/content/en/docs/reference/clusterspec/vsphere.md

* [Registry Mirror]({{< relref "optional/registrymirror.md" >}})

> 

Reviewer: I guess relrefs won't work in this context and we are going full html to do this?

Author: Yes. Once you put ````yaml` or equivalent around the content, you can't add links or other markup. We lose the "copy-to-clipboard" button, so someone would have to highlight to copy the whole text themselves. But I think they wouldn't need to do that since they would be working from their own generated yaml file.

            - 10.96.0.0/12
   controlPlaneConfiguration:        <a href="#controlplaneconfiguration-required"># Specific cluster control plane config (required)</a>
      count: 1                       <a href="#controlplaneconfigurationcount-required"># Number of control plane nodes (required)</a>
      endpoint:                      <a href="#controlplaneconfigurationendpointhost-required"># IP for control plane endpoint (required)</a>
         host: ""
      machineGroupRef:               <a href="#controlplaneconfigurationmachinegroupref-required"># vSphere-specific Kubernetes node config (required)</a>
        kind: VSphereMachineConfig
        name: my-cluster-machines
      taints:                        <a href="#controlplaneconfigurationtaints"># Taints applied to control plane nodes </a>
      - key: ""
        value: ""
        effect: ""
      labels:                        <a href="#controlplaneconfigurationlabels"># Labels applied to control plane nodes </a>
        "<key1>": ""
        "<key2>": "" 
   datacenterRef:                    <a href="#datacenterref"># Kubernetes object with vSphere-specific config </a>

>       kind: VSphereDatacenterConfig  <a href="#kubernetesversion-required"># Version of Kubernetes to use for cluster (required)</a>

Reviewer 2: ```suggestion
      kind: VSphereDatacenterConfig 
```

      name: my-cluster-datacenter
   externalEtcdConfiguration:
     count: 3                        <a href="#externaletcdconfigurationcount"># Number of etcd members </a>
     machineGroupRef:                <a href="#externaletcdconfigurationmachinegroupref"># vSphere-specific Kubernetes etcd config</a>
        kind: VSphereMachineConfig
        name: my-cluster-machines
   kubernetesVersion: "1.25"         <a href="#kubernetesversion-required"># Kubernetes version to use for the cluster (required)</a>
   workerNodeGroupConfigurations:    <a href="#workernodegroupconfigurations-required"># List of node groups you can define for workers (required) </a>
   - count: 1                        <a href="#workernodegroupconfigurationscount"># Number of worker nodes </a>
     machineGroupRef:                <a href="#workernodegroupconfigurationsmachinegroupref-required"># vSphere-specific Kubernetes node objects (required) </a>
       kind: VSphereMachineConfig
       name: my-cluster-machines
     name: md-0                      <a href="#workernodegroupconfigurationsname-required"># Name of the worker nodegroup (required) </a>

>      taints:                         <a href="#workernodegroupconfigurationsname-required"># Taints to apply to worker node group nodes </a>

Reviewer 2: ```suggestion
     taints:                         <a href="#workernodegroupconfigurationstaints"># Taints to apply to worker node group nodes </a>
```

Author: Another great catch. I knew I was going to mess up at least one of those.

Author: Fixed.

metadata:
   name: my-cluster-datacenter
spec:
  datacenter: ""                     <a href="#datacenter-required"># vSphere datacenter name on which to deploy EKS Anywhere (required) </a>
  disableCSI: false                  <a href="#disablecsi-optional"># Set to true to not have EKS Anywhere install and manage vSphere CSI driver</a>
  server: ""                         <a href="#server-required"># FQDN or IP address of vCenter server (required) </a>
  network: ""                        <a href="#network-required"># Path to the VM network on which to deploy EKS Anywhere (required) </a>
  insecure: false                    <a href="#insecure-optional"># Set to true if vCenter does not have a valid certificate </a>
  thumbprint: ""                     <a href="#thumbprint-required-if-insecurefalse"># SHA1 thumprint of vCenter server certificate (required if insecure=false)</a>

---
apiVersion: anywhere.eks.amazonaws.com/v1alpha1
kind: VSphereMachineConfig
metadata:
   name: my-cluster-machines
spec:

>   diskGiB:                          <a href="#diskgib-optional"># Size of disk on VMs, if no snapshots</a>

Reviewer 3: We updated EKS-A recently to make the disk size configurable regardless of snapshots.
```suggestion
  diskGiB:                          <a href="#diskgib-optional"># Size of disk on VMs</a>
```
Also, if you don't mind, can you remove that from [here](https://anywhere.eks.amazonaws.com/docs/reference/clusterspec/vsphere/#diskgib-optional)?

Author: I'll do that in a separate PR, since this already merged.
